### PR TITLE
Align class names and update AzureResourceType variable type to ResourceType

### PR DESCRIPTION
**Pull request by YalinLi0312**

### CODE REVIEW

**Comment by bquantump** [comment]
What is ADO item for this?

**Comment by YalinLi0312** [comment]
> What is ADO item for this?

4965

**Review by allenjzhang — APPROVED** [review]

**Review by markcowl — APPROVED** [review]

File: azure-proto-core/ResourceContainer.cs [hunk 1]
```diff
@@ -19,16 +19,16 @@ namespace azure_proto_core
     ///     The return type of the Creation methods, this can be Response<typeparamref name="T" /> or a
     ///     long-running response
     /// </typeparam>
-    public abstract class ResourceContainerOperations<TOperations, TResource> : OperationsBase
```
Comment (m-nash) [inline azure-proto-core/ResourceContainer.cs:22]: I think we want to name this ResourceContainerBase to match the pattern of ResourceOperationsBase

Its the concrete classes that need to follow the [ResourceName]Container pattern such as SubscriptionContainer or VirtualMachineContainer

**Review by m-nash — CHANGES_REQUESTED** [review]

File: azure-proto-core/ResourceContainerBase.cs [hunk 2]
```diff
@@ -19,16 +19,16 @@ namespace azure_proto_core
     ///     The return type of the Creation methods, this can be Response<typeparamref name="T" /> or a
     ///     long-running response
     /// </typeparam>
-    public abstract class ResourceContainerOperations<TOperations, TResource> : OperationsBase
+    public abstract class ResourceContainerBase<TOperations, TResource> : OperationsBase
```
Comment (m-nash) [inline azure-proto-core/ResourceContainerBase.cs:22]: When we change the class name we should change the file name to match

File: azure-proto-core/ResourceGroupContainer.cs [hunk 3]
```diff
@@ -14,12 +14,12 @@ namespace azure_proto_core
     /// Operations for the RespourceGroups container in the given subscription context.  Allows Creating and listign respource groups
     /// and provides an attachment point for Collections of Tracked Resources.
     /// </summary>
-    public class ResourceGroupContainerOperations : ResourceContainerOperations<ResourceGroup, ResourceGroupData>
+    public class ResourceGroupContainer : ResourceContainerBase<ResourceGroup, ResourceGroupData>
```
Comment (m-nash) [inline azure-proto-core/ResourceGroupContainer.cs:17]: change filename to match

File: azure-proto-core/SubscriptionContainer.cs [hunk 4]
```diff
@@ -11,11 +11,11 @@ namespace azure_proto_core
     /// <summary>
     /// Subscription Container Operationss
     /// </summary>
-    public class SubscriptionContainerOperations : OperationsBase
+    public class SubscriptionContainer : OperationsBase
```
Comment (m-nash) [inline azure-proto-core/SubscriptionContainer.cs:14]: change file name to match

**Review by m-nash — CHANGES_REQUESTED** [review]

File: azure-proto-core/ResourceContainerBase.cs (hunk 2 above)
Comment (YalinLi0312) [reply azure-proto-core/ResourceContainerBase.cs:22]: The file name is ResourceContainerBase.cs, it has aligned with the class name ResourceContainerBase. Same as rest.

**Review by m-nash — APPROVED** [review]

**Pull request merged**